client/cmd/dexc-desktop/pkg/pkg-darwin.sh

```diff
@@ -0,0 +1,334 @@
+	echo "Device name:     $DEV_NAME"
+
+
+	echo "Copying background file '$BACKGROUND_FILE'..."
+	[[ -d "$MOUNT_DIR/.background" ]] || mkdir "$MOUNT_DIR/.background"
+	cp "$BACKGROUND_FILE" "$MOUNT_DIR/.background/$BACKGROUND_FILE_NAME"
+
+	echo "Making link to Applications dir..."
+	ln -s /Applications "$MOUNT_DIR/Applications"
+
+	# Tell the volume that it has a special file attribute that will make it use the
+	# custom icon.
+	echo "Copying volume icon file '$VOLUME_ICON_FILE'..."
+	cp "$VOLUME_ICON_FILE" "$MOUNT_DIR/.VolumeIcon.icns"
+	SetFile -c icnC "$MOUNT_DIR/.VolumeIcon.icns"
+	SetFile -a C "$MOUNT_DIR"
+
+	# Run AppleScript to do all the Finder cosmetic stuff
+	sleep 2 # pause to workaround occasional "Can’t get disk" (-1728) issues
+	echo "Running AppleScript to make Finder stuff pretty..."
+	if /usr/bin/osascript "${APPLESCRIPT_FILE}" "${VOLUME_NAME}"; then
+		# Okay, we're cool
+		true
+	else
+		echo >&2 "Failed running AppleScript"
+		hdiutil_detach "${DEV_NAME}"
+		exit 64
+	fi
+
+	echo "Done running the AppleScript..."
+	sleep 4
+	rm "$APPLESCRIPT_FILE"
+
+	# Make sure it's not world writeable
+	echo "Fixing permissions..."
+	chmod -Rf go-w "${MOUNT_DIR}"
+	echo "Done fixing permissions"
+
+	# Delete unnecessary file system events log if possible
+	echo "Deleting .fseventsd"
+	rm -rf "${MOUNT_DIR}/.fseventsd"
+
+	hdiutil_detach "${DEV_NAME}"
+
+	# Compress image
+	echo "Compressing disk image..."
+	hdiutil convert -quiet "${TEMP_DMG}" -format ${FORMAT} ${IMAGEKEY} -o "${DMG_FILE}"
+	rm -f "${TEMP_DMG}"
+
+	# Check if hdiutil supports internet-enable, support was removed in macOS 10.15.
+	if hdiutil internet-enable -help >/dev/null 2>/dev/null; then
+		hdiutil internet-enable -yes "${DMG_FILE}"
+	else
+		echo "hdiutil does not support internet-enable. Note it was removed in macOS 10.15."
+	fi
+
+	echo "Disk image done"
+}
+
+# Build the webpack bundle prior to building the webserver package, which embeds
+# the files.
+if [[ -n "$1" ]]; then
+	pushd ../../../webserver/site
+	go generate # just check, no write
+	npm ci
+	npm run build
+	popd
+fi
+
+function build_targets() {
+  for TARGET in ${TARGETS}; do
+    OS=${TARGET%%/*}
+    ARCH=${TARGET##*/}
+
+    echo "Building .DMG click installer for ${OS}-${ARCH}"
+
+	TARGET_NAME="dexc${FLAVOR}-${OS}-${ARCH}-v${VER}"
+
+	# Remove any exiting executable if any.
+	rm -rf "${APP_EXCE_DIR}"
+	mkdir -p "${APP_EXCE_DIR}"
+
+    pushd ..
+    GOOS=${OS} GOARCH=${ARCH} go build -v -trimpath ${TAGS_DEXC:+-tags ${TAGS_DEXC}} -o "${APP_EXCE_DIR}/${APP_NAME}" -ldflags "${LDFLAGS_DEXC:-${LDFLAGS_BASE}}"
+    popd
+
+	createDMGInstaller "${TARGET_NAME}"
+
+  done
+}
+
+TARGETS="darwin/amd64" # unable to build for darwin/arm64 on a darwin/amd64 machine.
+FLAVOR="-tray"
+TAGS_DEXC="systray"
+prepare
+build_targets
+cleanup
+
+pushd ./installers
+sha256sum *.dmg > dexc-v${VER}-manifest.txt
```

Comment:
I think this `sha256sum` is just on linux. I have `shasum -a 256` on mac.

Reply:
Surprisingly, I have both.